subprojects/reporting/src/main/java/org/gradle/api/reporting/internal/SimpleReport.java

```diff
@@ -53,13 +60,23 @@ public String toString() {
     }
 
     public File getDestination() {
-        return destination == null ? null : resolveToFile(destination);
+        Object evaluatedDestination = destination.get();
+        return evaluatedDestination == null ? null : resolveToFile(evaluatedDestination);
     }
 
-    protected void setDestination(Object destination) {
+    public void setDestination(Provider<Object> destination) {
         this.destination = destination;
     }
 
+    public void setDestination(final Object destination) {
+        this.destination = project.provider(new Callable<Object>() {
```

Comment:
should replace this with a method that is a basically a lazy `Project.file()` method, that takes an object and returns a `Provider<File>`.